hal/armv8m/mpu.c

```diff
@@ -20,27 +20,170 @@
 static mpu_common_t mpu_common;
 
 
+enum {
+	mpu_type,
+	mpu_ctrl,
+	mpu_rnr,
+	mpu_rbar,
+	mpu_rlar,
+	mpu_rbar_a1,
+	mpu_rlar_a1,
+	mpu_rbar_a2,
+	mpu_rlar_a2,
+	mpu_rbar_a3,
+	mpu_rlar_a3,
+	mpu_mair0 = 0xC,
+	mpu_mair1
+};
+
+
+/* Translate memory map attributes to RLAR attribute bits */
+static u32 mpu_regionAttrsRlar(u32 attr, unsigned int enable)
+{
+	u8 attrIndx = 0, execNever = 0;
+	attrIndx |= ((attr & mAttrCacheable) != 0) ? 1 : 0;
+	attrIndx |= ((attr & mAttrBufferable) != 0) ? (1 << 1) : 0;
+	execNever |= ((attr & mAttrExec) == 0) ? 1 : 0;
+	return (execNever << 4) | (attrIndx << 1) | ((enable != 0) ? 1 : 0);
+}
+
+
+/* Translate memory map attributes to RBAR attribute bits */
+static u32 mpu_regionAttrsRbar(u32 attr, unsigned int enable)
+{
+	u8 ap = 0; /* set privileged read-write access, unprivileged no access */
+
+	if ((attr & mAttrRead) != 0) {
+		/*
+		 * set privileged read-write access, unprivileged read only access
+		 * armv8m does not support that
+		 * instead set both levels to read/write
```

Comment:
I suggest a wording change:
```suggestion
		 * TODO: Privileged read/write and unprivileged read/only configuration is not supported
		 * on ARMv8-M. As a workaround for now we treat it as read/write for both privilege levels.
```